meson-mode.el

```diff
@@ -900,34 +900,66 @@ arguments."
 	    (fspec (alist-get fname meson-builtin-functions)))
       (plist-get fspec :doc)))
 
-(defun meson-lookup-doc (what)
-  "Open Meson reference manual and find heading starting with WHAT."
-  (when-let (refman (seq-find 'file-exists-p
-			      (mapcar (lambda (file) (expand-file-name file meson-markdown-docs-dir))
-				      '("Reference-manual.md"
-					"Reference-manual.md.gz"))))
-    (find-file-read-only refman)
-    (when (and (fboundp 'markdown-view-mode)
-	       (not (eq major-mode 'markdown-view-mode)))
-      (markdown-view-mode))
-    (local-set-key (kbd "q") 'bury-buffer)
-    (goto-char (point-min))
-    (re-search-forward (concat "^#+ " what))
-    (recenter 0)))
-
-(defun meson-lookup-doc-at-point ()
-  "Show Meson documentation related to current point.
-Currently, it shows something only for functions."
-  (interactive)
-  (if-let ((func (meson-function-at-point)))
-      (meson-lookup-doc func)
-    (message "Nothing to look up")))
+(defun meson--find-reference-manual ()
+  "Return the absolute filename of the Meson reference manual or nil."
+  (let ((default-directory meson-markdown-docs-dir))
+    (when-let (manual (seq-find #'file-exists-p '("Reference-manual.md"
+                                                  "Reference-manual.md.gz")))
+      (expand-file-name manual))))
+
+(defun meson--make-lookup-regexp (identifier)
+  "Make regexp for looking up IDENTIFIER in the Meson reference manual."
+  ;; In Emacs 27 this could be simplified to (rx ... (literal identifier) ...).
+  (rx-to-string
+   `(seq bol (or (seq (+ "#") " " (? "`") ,identifier (or "(" "`" eol))
+                 (seq "- `" ,identifier "(")))))
```

Comment:
I've found that for matching methods of dependency objects, the following change is needed:
```suggestion
                 (seq (* blank) "- `" ,identifier "(")))))
```